api/src/main/java/javax/mvc/engine/ViewEngineContext.java

```diff
@@ -74,21 +71,6 @@
     Models getModels();
 
     /**
-     * Returns HTTP request object from the Servlet container.
-     *
-     * @return HTTP request object.
-     */
-    HttpServletRequest getRequest();
-
-    /**
-     * Returns HTTP response object from the servlet container. The underlying
-     * output stream should be used to write the result of processing a view.
-     *
-     * @return HTTP response object.
-     */
-    HttpServletResponse getResponse();
-
```

Comment:
Regarding the header API: I personally don't like the mutable `MultivaluedMap` provided to the `MessageBodyWriter`. Providing a mutable map as a parameter feels weird to me. So I would prefer something like `ViewEngineContext#setHeader(String, String)`.

BTW: The header map in JAX-RS is defined as `MultivaluedMap<String,**Object**>`. Anyone knows why this is not `String`?

Regarding the encoding: IMO the view engine is responsible to set the correct encoding header. That brings us back to the discussion Florian started regarding the different ways to specify the encoding. Like `@Produces` vs. setting it in the JSP. But IMO this should be the job of the view engine. `@Produces` should only be used for the matching algorithm.